examples/terraform/aws-dualstack/main.tf

```
@@ -0,0 +1,414 @@
+}
+
+############################### NETWORKING SETUP ###############################
+
+resource "aws_subnet" "public" {
+  count                   = 3
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  map_public_ip_on_launch = true
+  vpc_id                  = data.aws_vpc.selected.id
+
+  cidr_block = cidrsubnet(
+    data.aws_vpc.selected.cidr_block,
+    local.subnet_newbits,
+    (random_integer.cidr_block.result + count.index) % local.subnet_total,
+  )
+
+  ipv6_cidr_block = cidrsubnet(
+    data.aws_vpc.selected.ipv6_cidr_block,
+    local.ipv6_subnet_newbits,
+    (random_integer.ipv6_cidr_block.result + count.index) % local.ipv6_subnet_total,
+  )
+
+  assign_ipv6_address_on_creation = true
+
+  tags = tomap({
+    "Name"                   = "${var.cluster_name}-${data.aws_availability_zones.available.names[count.index]}",
+    "Cluster"                = var.cluster_name,
+    (local.kube_cluster_tag) = "shared",
+  })
+}
+
+################################### FIREWALL ###################################
+
+resource "aws_security_group" "common" {
+  name        = "${var.cluster_name}-common"
+  description = "cluster common rules"
+  vpc_id      = data.aws_vpc.selected.id
+
+  tags = tomap({
+    "Cluster"                = var.cluster_name,
+    (local.kube_cluster_tag) = "shared",
+  })
+}
+
+resource "aws_security_group_rule" "ingress_self_allow_all" {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all incoming traffic from members of this group"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  self        = true
+}
+
+resource "aws_security_group_rule" "egress_allow_all" {
+  type              = "egress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all outgoing traffic"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "nodeports" {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "open nodeports"
+  from_port   = 30000
+  to_port     = 32767
+  protocol    = "tcp"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group" "elb" {
+  name        = "${var.cluster_name}-api-lb"
+  description = "kube-api firewall"
+  vpc_id      = data.aws_vpc.selected.id
+
+  egress {
+    description = "allow all outgoing traffic"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "allow anyone to connect to tcp/6443"
+    from_port   = 6443
+    to_port     = 6443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    description = "allow all outgoing traffic"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    description = "allow anyone to connect to tcp/6443"
```

Let's mention in the description that this is an IPv6 rule.

```suggestion
    description = "allow anyone to connect to tcp/6443 over ipv6"
```